src/battle-dex.ts

```diff
@@ -851,15 +851,16 @@ class ModdedDex {
 
 			let data = {...Dex.moves.get(name)};
 
-			const table = window.BattleTeambuilderTable[this.modid];
-			if (id in table.overrideAcc) data.accuracy = table.overrideAcc[id];
-			if (id in table.overrideBP) data.basePower = table.overrideBP[id];
-			if (id in table.overridePP) data.pp = table.overridePP[id];
-			if (id in table.overrideMoveType) data.type = table.overrideMoveType[id];
-			if (id in table.overrideMoveCategory) data.category = table.overrideMoveCategory[id];
-			if (id in table.overrideMoveDesc) data.desc = table.overrideMoveDesc[id];
-			if (id in table.overrideMoveShortDesc) data.shortDesc = table.overrideMoveShortDesc[id];
-
+			const overrideMoveKeys = ['accuracy', 'basePower', 'category', 'desc', 'flags', 'pp', 'shortDesc', 'target', 'type'];
+			for (const key of overrideMoveKeys) {
+				for (var i = this.gen; i < 8; i++) {
+					const table = window.BattleTeambuilderTable['gen' + i];
+					if (id in table.overrideMoveData && key in table.overrideMoveData[id]) {
+						data[key] = table.overrideMoveData[key];
+						break;
+					}
+				}
+			}
```

Comment:
```suggestion
			for (let i = Dex.gen - 1; i >= this.gen; i--) {
				const table = window.BattleTeambuilderTable[`gen${i}`];
				if (id in table.overrideMoveData) {
					Object.assign(data, table.overrideMoveData[id]);
				}
			}
```

Surprised the linter isn't complaining about the `var`.

Anyway, here, this code passes TypeScript.

Reply:
It's also more generic (doesn't require updating this if we change what properties we add to `overrideMoveData`). I don't even think the perf difference is noticeable.

Reply:
Remember to do the same thing for Species.